annotation/tests/test_tasks_services.py

```diff
@@ -1,30 +1,240 @@
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
+    metric.task_from = datetime(2024, 1, 1)
+    metric.task_to = datetime(2024, 2, 1)
+    metric.agreement_metric = True
+    yield metric
+
+
+@pytest.fixture
+def mock_stats(mock_task: ManualAnnotationTask, mock_metric: Mock):
+    stat1 = MagicMock()
+    stat1.task = mock_task
+    stat1.task_id = 1
+    stat1.created = datetime(2024, 1, 1, 12, 0, 0)
+    stat1.updated = datetime(2024, 1, 2, 12, 0, 0)
+    stat1.task.user_id = 2
+    stat1.task.file_id = 3
+    stat1.task.pages = [1, 2, 3]
+    stat1.task.status.value = "completed"
+    stat1.task.agreement_metrics = [mock_metric]
+
+    stat2 = MagicMock()
+    stat2.task = mock_task
+    stat2.task_id = 2
+    stat2.created = datetime(2024, 1, 3, 12, 0, 0)
+    stat2.updated = datetime(2024, 1, 4, 12, 0, 0)
+    stat2.task.user_id = 2
+    stat2.task.file_id = 3
+    stat2.task.pages = [1, 2, 3]
+    stat2.task.status.value = "completed"
+    stat2.task.agreement_metrics = [mock_metric]
+
+    stat3 = MagicMock()
+    stat3.task = mock_task
+    stat3.task_id = 3
+    stat3.created = datetime(2024, 1, 5, 12, 0, 0)
+    stat3.updated = datetime(2024, 1, 6, 12, 0, 0)
+    stat3.task.user_id = 2
+    stat3.task.file_id = 3
+    stat3.task.pages = [1, 2, 3]
+    stat3.task.status.value = "completed"
+    stat3.task.agreement_metrics = [mock_metric]
+
+    yield [stat1, stat2, stat3]
+
+
+@pytest.fixture
+def mock_db(mock_stats):
+    mock_db = MagicMock()
+    mock_query = MagicMock()
+    mock_query.filter.return_value.filter.return_value.all.return_value = (
+        mock_stats
+    )
+    mock_db.query.return_value = mock_query
+    return mock_db
+
+
+@pytest.fixture
+def mock_task():
+    task = MagicMock(ManualAnnotationTask)
+    task.id = 1
+    task.user_id = 2
+    task.job_id = 10
+    task.file_id = 3
+    task.pages = {1, 2, 3}
+    task.is_validation = False
+    yield task
+
+
+@pytest.fixture
+def mock_tenant_data():
+    mock_data = MagicMock(TenantData)
+    mock_data.token = "mock_token"
+    yield mock_data
+
+
+@pytest.fixture
+def response_scores():
+    return [
+        ResponseScore(task_id=2, agreement_score=0.9),
+        ResponseScore(task_id=3, agreement_score=0.7),
+        ResponseScore(task_id=2, agreement_score=0.9),
+    ]
+
+
+@pytest.fixture
+def mock_parse_obj_as():
+    with patch("pydantic.parse_obj_as") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_get_unique_scores():
+    with patch(
+        "annotation.tasks.services.get_unique_scores", return_value=None
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_task_metric():
+    with patch("annotation.tasks.services.TaskMetric") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_agreement_score_comparing_result():
+    with patch(
+        "annotation.tasks.services.AgreementScoreComparingResult"
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_agreement_score_response():
```

Comment:
any reason using mock here? why not the real model?

Reply:
Ok:)